parser.y

```diff
@@ -253,14 +253,16 @@ rule
   rule_args: IDENTIFIER { result = [val[0]] }
            | rule_args "," IDENTIFIER { result = val[0].append(val[2]) }
 
-  rule_rhs_list: rule_rhs
+  rule_rhs_list: rule_rhs if_clause?
                 {
                   builder = val[0]
+                  builder.symbols << val[1] if val[1]
                   result = [builder]
                 }
-          | rule_rhs_list "|" rule_rhs
+          | rule_rhs_list "|" rule_rhs if_clause?
                 {
                   builder = val[2]
+                  builder.symbols << val[3] if val[3]
```

Comment:
`builder` (is the instance of `Grammar::ParameterizingRule::Rhs`) has one or zero `if_clause`. Then I think it's reasonable for `builder` has attribute for `if_clause` than holding `if_clause` on `symbols`.